src/useMutation.ts

```diff
@@ -6,13 +6,29 @@ import useMounted from '@restart/hooks/useMounted';
 import {
     MutationNode,
     MutationConfig,
+    MutationConfigWithoutVariables,
+    MutationConfigWithVariables,
     MutationState,
     Mutate,
-    MutationProps,
+    MutateWithVariables,
+    MutateWithoutVariables,
 } from './RelayHooksTypes';
 import { useRelayEnvironment } from './useRelayEnvironment';
+
 const { useCallback, useState } = React;
 
+export function useMutation<T extends MutationParameters>(
+    mutation: MutationNode<T>,
+    userConfig?: MutationConfigWithoutVariables<T>,
+    /** if not provided, the context environment will be used. */
+    environment?: Environment,
+): [MutateWithVariables<T>, MutationState<T>];
+export function useMutation<T extends MutationParameters>(
+    mutation: MutationNode<T>,
+    userConfig?: MutationConfigWithVariables<T> ,
+    /** if not provided, the context environment will be used. */
+    environment?: Environment,
+): [MutateWithoutVariables<T>, MutationState<T>];
```

Comment:
change `): [MutateWithoutVariables<T>, MutationState<T>];` to `): [Mutate<T>, MutationState<T>];`

This allows you to give users more flexibility